### ISSUE TITLE
No info about SC1091 rule in README - Rules

### ISSUE THREAD

**Issue by softwaredevelop** [hadolint/hadolint#818]
-   [x] This is a bug report
-   [ ] This is a feature request
-   [x] I searched existing issues before opening this one - #624

### Expected behavior

Info about SC1091 rule in README - Rules

### Actual behavior

No info about SC1091 in README - Rules but hadolint ignored SC1091

### Steps to reproduce the behavior

**Output of `hadolint --version` or
`docker run --rm hadolint/hadolint hadolint --version` or
`docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
hadolint --version
Haskell Dockerfile Linter 2.10.0
```

**Dockerfile** (if relevant)

```Dockerfile
ARG VARIANT=focal
FROM docker.io/library/ubuntu:$VARIANT

RUN apt-get update \
    && apt-get install --no-install-recommends --assume-yes \
    sudo \
    locales \
    ca-certificates \
    curl \
    zip \
    unzip && \
    echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen && \
    locale-gen && \
    apt-get clean && rm -fr /var/lib/apt/lists/*

ARG USERNAME=vscode
ARG USER_UID=1000
ARG USER_GID=$USER_UID

RUN groupadd --gid $USER_GID $USERNAME \
    && useradd --gid $USER_GID \
    --create-home $USERNAME \
    --shell /bin/bash \
    --uid $USER_UID \
    && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME \
    && chmod 0440 /etc/sudoers.d/$USERNAME


SHELL ["/bin/bash", "-o", "pipefail", "-c"]
RUN apt-get update \
    && apt-get install --no-install-recommends --assume-yes \
    curl \
    && curl -s "https://get.sdkman.io" | bash \
    # shellcheck source $HOME/.sdkman/bin/sdkman-init.sh
    && source "$HOME/.sdkman/bin/sdkman-init.sh" \
    && apt-get clean && rm -fr /var/lib/apt/lists/*
```

**.hadolint.yaml**

```yml
---
ignored:
    - DL3008
    - SC1091
```

**Additional environment details (OS, stack version, etc.)**

-   OS: Debian GNU/Linux 11 (bullseye)
-   Haskell Dockerfile Linter: prebuilt binary, hadolint-Linux-x86_64
-   Editor: Visual Studio Code 1.66.1
-   VS Code hadolint extension (michaellzc): v1.1.1
-   Visual Studio Code **settings.json**

```json
  "hadolint.hadolintPath": "/usr/local/bin/hadolint",
  "hadolint.cliOptions": [
    "--no-color"
  ],
```

-   Visual Studio Code problem message:

```bash
Value must be "DL3000" | "DL3001" | "DL3002" | "DL3003" | "DL3004" | "DL3005" | "DL3006" | "DL3007" | "DL3008" | "DL3009" | "DL3010" | "DL3011" | "DL3012" | "DL3013" | "DL3014" | "DL3015" | "DL3016" | "DL3018" | "DL3019" | "DL3020" | "DL3021" | "DL3022" | "DL3023" | "DL3024" | "DL3025" | "DL3026" | "DL3027" | "DL3028" | "DL3029" | "DL3030" | "DL3032" | "DL3033" | "DL3034" | "DL3035" | "DL3036" | "DL3037" | "DL3038" | "DL3040" | "DL3041" | "DL3042" | "DL3043" | "DL3044" | "DL3045" | "DL3046" | "DL3047" | "DL3048" | "DL3049" | "DL3050" | "DL3051" | "DL3052" | "DL3053" | "DL3054" | "DL3055" | "DL3056" | "DL3057" | "DL3058" | "DL3059" | "DL3060" | "DL4000" | "DL4001" | "DL4003" | "DL4004" | "DL4005" | "DL4006" | "SC1000" | "SC1001" | "SC1007" | "SC1010" | "SC1018" | "SC1035" | "SC1045" | "SC1065" | "SC1066" | "SC1068" | "SC1077" | "SC1078" | "SC1079" | "SC1081" | "SC1083" | "SC1086" | "SC1087" | "SC1095" | "SC1097" | "SC1098" | "SC1099" | "SC2002" | "SC2015" | "SC2026" | "SC2028" | "SC2035" | "SC2039" | "SC2046" | "SC2086" | "SC2140" | "SC2154" | "SC2155" | "SC2164".
```

**Comment by softwaredevelop** [comment]
sdkman & source command & vscode remote containers & podman are very tricky in Containerfile 🌤😅

```Dockerfile
ARG USERNAME=vscode
ARG USER_UID=1000
ARG USER_GID=$USER_UID

RUN \
    groupadd --gid $USER_GID $USERNAME; \
    useradd --gid $USER_GID \
    --create-home $USERNAME \
    --shell /bin/bash \
    --uid $USER_UID; \
    echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME; \
    chmod 0440 /etc/sudoers.d/$USERNAME

USER $USERNAME

SHELL ["/bin/bash", "-o", "pipefail", "-c"]
RUN \
    curl --silent "https://get.sdkman.io" | bash

RUN \
    /bin/bash -c \
    "source $HOME/.sdkman/bin/sdkman-init.sh; \
    sdk install java 17.0.2-ms; \
    sdk install gradle 7.4.2; \
    rm -rf $HOME/.sdkman/archives/*; \
    rm -rf $HOME/.sdkman/tmp/*"
```

**Comment by Abdillah** [comment]
What makes this issue solved?

**Comment by Hongbo-Miao** [comment]
I have same question, created a ticket at https://github.com/hadolint/hadolint/issues/905

**Comment by softwaredevelop** [comment]
This is a shellchek problem. Use quotation mark.
```Dockerfile
RUN \
    /bin/bash -c \
    "source $HOME/.sdkman/bin/sdkman-init.sh; \
    sdk install java 17.0.2-ms; \
    sdk install gradle 7.4.2; \
    rm -rf $HOME/.sdkman/archives/*; \
    rm -rf $HOME/.sdkman/tmp/*"
```
